website/src/_posts/2022-03-2.4-2.7.md

```diff
@@ -0,0 +1,72 @@
+---
+title: "Uppy 2.4-2.7: Compress Images, Transloadit Rate Limiting, Move to ESM" 
+date: 2022-03-11
+author: 
+  - aduh95
+  - arturi
+  - mifi
+  - murderlon
+image: 
+published: true
+---
+
+Recent Uppy updates brings image compression, improved Transloadit rate limiting, lighter Dashboard. We’ve also moved our end to end tests to Cypress, and are slowly converting to ESM modules.
+
+<img class="border" alt="Uppy Compressor plugin showing compressed images notification" src="/images/blog/2.4-2.7/compressor.jpg">
+
+<!--more-->
+
+## Compressor
+
+Some of the most commonly uploaded files on the internet are images — photos, stock images, screenshots, documents scans. Uppy now includes a new plugin, `@uppy/compressor` that optimizes images (JPEG, PNG), saving on average up to 60% in size (roughly 18 MB for 10 images). It uses [Compressor.js](https://github.com/fengyuanchen/compressorjs) under the hood.
+
+[Read the docs](https://uppy.io/docs/compressor/) for options and try it out.
+
+```js
+import Uppy from '@uppy/core'
+import Compressor from '@uppy/compressor'
+
+const uppy = new Uppy()
+uppy.use(Compressor)
+```
+
+`@uppy/compressor` is also the first plugin that we are releasing as ESM, backwards compatible as CJS, to test out how this works. Please report any issues.
+
+## Hosted Companion Custom Credentials
+
+It’s no secret [Transloadit hosted Companion](https://transloadit.com/docs/sdks/uppy/) is the easiest secure way to allow your users pick files from cloud sources like Google Drive, Instagram, Box, etc via Uppy. Up until now, however, that meant your users had to authenticate their cloud account with Transloadit, while you might want to show your own logo instead. Now you can do just that, but heading to `Transloadit Console` / `Credentials`, scroll to `Third-party Credentials`, click on `Add new Credentials` and select `Companion OAuth`.
+
+<img class="border" alt="Transloadit Companion 3rd party credentials" src="/images/blog/2.4-2.7/companion-3rd-party-oauth.jpg">
+
+Try it out with [Free Transloadit Community plan — 5GB / month and Companion](https://transloadit.com/pricing/).
+
+## Transloadit Rate Limiting
+
+We’ve added rate limiting to Transloadit [assembly creation and status polling](https://github.com/transloadit/uppy/pull/3429) and [tus retries](https://github.com/transloadit/uppy/pull/3394):
+
+> When the remote server responds with HTTP 429, all requests are paused for a while in the hope that it can resolve the rate limiting. Failed requests are also now queued up after the retry delay. Before that, they were simply scheduled which would sometimes end up overflowing the limit option.
```

Comment:
```suggestion
> When the remote server responds with HTTP 429, all requests are paused for a while in the hope that it can resolve the rate limiting. Failed requests are also now queued up after the retry delay. Before that, they were scheduled for retry without being queued up, which would sometimes end up overflowing the limit option.
```